spec/requests/api/v1/participant_declarations_spec.rb

```diff
@@ -141,61 +140,133 @@ def build_params(attributes)
           ect_profile.participant_profile_states.create({ state: "deferred", created_at: declaration_date - 1.second })
         end
 
-        it "returns 422" do
+        it "returns 200" do
           params = build_params(valid_params)
           post "/api/v1/participant-declarations", params: params
-          expect(response.status).to eq 422
-          expect(response.body).to eq({ errors: [{ title: "Bad or missing parameters", detail: I18n.t(:declaration_on_incorrect_state) }] }.to_json)
+          expect(response.status).to eq 200
         end
       end
 
       context "when the participant transfers to a new school with a different lead provider" do
         let(:new_programme) { create(:induction_programme, :fip) }
         let(:transfer_lp_token) { LeadProviderApiToken.create_with_random_token!(cpd_lead_provider: new_programme.partnership.lead_provider.cpd_lead_provider) }
         let(:transfer_lp_bearer_token) { "Bearer #{transfer_lp_token}" }
+        let(:url) { "/api/v1/participants/ecf/#{ect_profile.user.id}/withdraw" }
+        let(:params) { { data: { attributes: { course_identifier: "ecf-induction", reason: "moved-school" } } } }
 
-        before do
-          induction_record.leaving!(ect_profile.schedule.milestones.first.start_date + 1)
-          Induction::Enrol.call(participant_profile: ect_profile, induction_programme: new_programme, start_date: ect_profile.schedule.milestones.first.start_date + 1)
-        end
+        context "when the participant has been withdrawn" do
+          before do
+            induction_record.leaving!(ect_profile.schedule.milestones.find_by(declaration_type: "started").start_date + 1)
+            Induction::Enrol.call(participant_profile: ect_profile, induction_programme: new_programme, start_date: ect_profile.schedule.milestones.find_by(declaration_type: "started").start_date + 1)
+            put url, params: build_params(params)
+          end
 
-        it "is possible for new lead provider to post a declaration" do
-          default_headers[:Authorization] = transfer_lp_bearer_token
+          it "is possible for new lead provider to post a declaration" do
+            default_headers[:Authorization] = transfer_lp_bearer_token
+            updated_params = valid_params.merge({ declaration_date: (ect_profile.schedule.milestones.find_by(declaration_type: "started").start_date + 2).rfc3339 })
```

Comment:
```suggestion
            updated_params = valid_params.merge({ declaration_date: (ect_profile.schedule.milestones.find_by(declaration_type: "started").start_date).rfc3339 })
```

Reply:
Would you mind extracting that `(ect_profile.schedule.milestones.find_by(declaration_type: "started").start_date` into a `let(:declaration_date)` ?

Reply:
👌